libs/market-depth/src/lib/orderbook.tsx

```diff
@@ -262,6 +270,50 @@ export const Orderbook = ({
     handleResize();
     return () => window.removeEventListener('resize', handleResize);
   }, []);
+  // sets the correct width of header and footer
+  useLayoutEffect(() => {
+    if (
+      !gridElement.current ||
+      !headerElement.current ||
+      !footerElement.current
+    ) {
+      return;
+    }
+    const gridWidth = gridElement.current.clientWidth;
+    headerElement.current.style.width = `${gridWidth}px`;
+    footerElement.current.style.width = `${gridWidth}px`;
+  }, [headerElement, footerElement, gridElement]);
+  // handles resizing of the Allotment.Pane (x-axis)
+  // adjusts the header and footer width
+  const gridResizeHandler: ResizeObserverCallback = useCallback(
+    (entries) => {
+      if (
+        !headerElement.current ||
+        !footerElement.current ||
+        entries.length === 0
+      ) {
+        return;
+      }
+      const {
+        contentRect: { width, height },
+      } = entries[0];
+      headerElement.current.style.width = `${width}px`;
+      footerElement.current.style.width = `${width}px`;
+      setViewportHeight(height);
```

Comment:
suggestion (non blocking): I realise that this component already uses `viewportHeight` and `setViewportHeight`, but perhaps its worth looking into using `AutoSizer`. We already wrap each trade grid window with it (TradeGridChild) and it should be fairly easy to refactor so that width and height are passed down as props

Reply:
I've created a new issue with your suggestion: https://github.com/vegaprotocol/frontend-monorepo/issues/1533
Wrapping the orderbook container quickly with AutoSizer resulted with width and height set to 0 😢. I think there's a clash somewhere or they don't like to be nested.

The change in width here is only needed to accommodate the scrollbar size so the rows are nicely aligned with each other and the scrollbar is not covered.